src/firmware/shared/circular_buffer_test.cpp

```diff
@@ -0,0 +1,85 @@
+extern "C"
+{
+#include "firmware/shared/circular_buffer.h"
+
+#include "shared/constants.h"
+}
+
+#include <gtest/gtest.h>
+
+class CircularBufferTest : public testing::Test
+{
+   protected:
+    virtual void SetUp(void)
+    {
+        circular_buffer = circular_buffer_create(CIRCULAR_BUFFER_MAX_NUM_ELEMENTS);
+    }
+
+    virtual void TearDown(void)
+    {
+        circular_buffer_destroy(circular_buffer);
+    }
+
+    CircularBuffer_t* circular_buffer;
+};
+
```

Comment:
Would be good to add tests for
* accessing a value at index `0`
* accessing a value at a negative index
* checking we hit the assert if the index is larger than the buffer size (can use `ASSERT_DEATH`)
* checking we hit the assert if the buffer is not full and the index tries to access an unpopulated value/index

Reply:
Added tests for:
- accessing at index 0
- checking assert if index > buffer size
- checking unpopulated value

Since `circular_buffer_getAtIndex` takes in a `uint8_t` now, it wouldn't make sense to test a negative index, right?

Reply:
Correct re: negative indices